site/content/en/docs/examples/hpa.md

```diff
@@ -0,0 +1,384 @@
+            limits:
+              cpu: 500m
+              memory: 256Mi
+```
+
+### Step 2: Create HPA
+
+Apply the HPA configuration targeting the LeaderWorkerSet:
+
+```yaml
+apiVersion: autoscaling/v2
+kind: HorizontalPodAutoscaler
+metadata:
+  name: lws-hpa
+spec:
+  minReplicas: 2
+  maxReplicas: 8
+  metrics:
+  - type: Resource
+    resource:
+      name: cpu
+      target:
+        type: Utilization
+        averageUtilization: 50
+  scaleTargetRef:
+    apiVersion: leaderworkerset.x-k8s.io/v1
+    kind: LeaderWorkerSet
+    name: leaderworkerset-sample
+```
+
+### Step 3: Verify HPA Setup
+
+Check that HPA is properly configured:
+
+```shell
+kubectl get hpa
+```
+
+Output should look similar to:
+```shell
+NAME      REFERENCE                         TARGETS   MINPODS   MAXPODS   REPLICAS   AGE
+lws-hpa   LeaderWorkerSet/leaderworkerset-sample   1%/50%    2         8         2          2m
+```
+
+Check the LeaderWorkerSet status for HPA pod selector:
+
+```shell
+kubectl get leaderworkerset leaderworkerset-sample -o jsonpath='{.status.hpaPodSelector}'
+```
+
+## Testing Autoscaling
+
+To test the autoscaling behavior, you can generate load on the leader pods:
+
+### Generate CPU Load
+
+⚠️ **Important**: HPA only monitors LeaderWorkerSet's leader pods, so you must generate CPU load in these pods to trigger scaling.
+
+```shell
+# Generate high CPU load in both leader pods
+kubectl exec leaderworkerset-sample-0 -- /bin/sh -c "for i in \$(seq 1 4); do yes > /dev/null & done"
+kubectl exec leaderworkerset-sample-1 -- /bin/sh -c "for i in \$(seq 1 4); do yes > /dev/null & done"
+```
+
+### Monitor Scaling
+
+Run the following commands in multiple terminals simultaneously to observe the scaling process:
+
+```shell
+# Monitor HPA status
+kubectl get hpa -w
+
+# Monitor LeaderWorkerSet replica changes
+kubectl get leaderworkerset leaderworkerset-sample -w
+
+# Monitor pods status
+kubectl get pods -l leaderworkerset.sigs.k8s.io/name=leaderworkerset-sample -w
+
+# Check detailed status and events
+kubectl describe hpa lws-hpa
+kubectl top pods -l leaderworkerset.sigs.k8s.io/name=leaderworkerset-sample
+```
+
+**Expected Scaling Behavior**
+
+After generating load, you should see within 2-3 minutes:
+1. CPU usage rises from 1% to 300%+
+2. HPA status changes from `1%/50%` to `300%/50%`
+3. LeaderWorkerSet replicas gradually increase to 8
+4. New leader and worker pods are created
+
+**Stop Load Testing**
+
+```shell
+# Restart leader pods to clear CPU load (nginx container doesn't have pkill command)
+kubectl delete pod leaderworkerset-sample-0 leaderworkerset-sample-1
+```
+
+After clearing the load, you will see within 5-10 minutes:
+1. CPU usage decreases to 0%
+2. HPA waits for the stabilization window before starting to scale down
+3. LeaderWorkerSet replicas gradually decrease to 2
+
+## Advanced HPA Configuration
+
+### Memory-based Scaling
+
+You can also configure HPA to scale based on memory utilization:
+
```

Comment:
HI @LuyuZhang00 

Considering other examples, their YAML files are placed in the https://github.com/kubernetes-sigs/lws/tree/main/docs/examples/sample directory rather than within the markdown files.

Should we create an HPA directory under docs/examples and place the YAML files there? This would make maintenance easier.